docs/install/build_from_source.md

```diff
@@ -1,93 +1,83 @@
 # Build MXNet from Source
 
-This document explains how to build MXNet from source code. Building MXNet from source is a two step process.
-
-1. Build the MXNet shared library, `libmxnet.so`, from [C++ source files](#build-the-shared-library)
-2. Install the [language bindings](#installing-mxnet-language-bindings) for MXNet. MXNet supports the following languages:
-    - Python
-    - C++
-    - Clojure
-    - Julia
-    - Perl
-    - R
-    - Scala
+This document explains how to build MXNet from source code.
+
+
+## Overview
+
+Building from source follows this general two-step flow of building the shared library, then installing your preferred language binding. Use the following links to jump to the different sections of this guide.
+
+1. Build the MXNet shared library, `libmxnet.so`.
+    * [Clone the repository](#clone-the-mxnet-project)
+    * [Prerequisites](#prerequisites)
+        * [Math library selection](#math-library-selection)
+        * [Install GPU software](#install-gpu-software)
+        * [Install optional software](#install-optional-software)
+    * [Adjust your build configuration](#build-configurations)
+    * [Build MXNet](#build-mxnet)
+        * [with NCCL](#build-mxnet-with-nccl) (optional)
+        * [for C++](#build-mxnet-with-c++) (optional)
+        * [Usage Examples](#usage-examples)
+            * [systems with GPUs and Intel CPUs](#recommended-for-Systems-with-NVIDIA-GPUs-and-Intel-CPUs)
+            * [GPUs with non-Intel CPUs](#recommended-for-Systems-with-Intel-CPUs)
+            * [Intel CPUs](#recommended-for-Systems-with-Intel-CPUs)
+            * [non-Intel CPUs](#recommended-for-Systems-with-non-Intel-CPUs)
+2. [Install the language API binding(s)](#installing-mxnet-language-bindings) you would like to use for MXNet.
+MXNet's newest and most popular API is Gluon. Gluon is built into the Python binding. If Python isn't your preference, you still have more options. MXNet supports several other language APIs:
+    - [Python (includes Gluon)](../api/python/index.html)
```

Comment:
The flow does not seem right. We are taking about Installing API bindings and we redirect the users to API page, we should redirect them to the language install page instead.